### PR TITLE
Optimize w8a8 kernel

### PR DESCRIPTION
**Pull request by grimoire**
Kernel would not recompile when M changes.

The performance on small batch size and short context is still not fast enough. Since triton kernel launch takes too much time.

### CODE REVIEW

File: lmdeploy/pytorch/kernels/w8a8_triton_kernels.py [hunk 1]
```diff
@@ -340,7 +311,7 @@ def _per_token_quant_int8(
     # Quant
     _absmax = tl.maximum(tl.max(tl.abs(y)), eps)
     y_s = _absmax / 127
-    y_q = tl.maximum(tl.minimum(tl.math.round(y / y_s), 127), -128).to(tl.int8)
+    y_q = tl.math.round(y / y_s).to(tl.int8)
```
Comment (AllentDan) [inline lmdeploy/pytorch/kernels/w8a8_triton_kernels.py:314]: shall we add assertion here to make sure `tl.math.round(y / y_s)` is in [-128, 127]

**Comment by AllentDan** [comment]
Does the optimization gain comes from avoiding triton kernel launch time? The triton kernel op actually slows down. What is the real model execution gain?

**Review by HIT-cwh — APPROVED** [review]

File: lmdeploy/pytorch/kernels/w8a8_triton_kernels.py (hunk 1 above)
Comment (grimoire) [reply lmdeploy/pytorch/kernels/w8a8_triton_kernels.py:314]: y_s = absmax(y) / 127. the value should be ok.
And it is not legal to add assert in triton kernel.

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**